e2e/tests/explore/explore.spec.ts

```diff
@@ -1,56 +1,143 @@
+      .findOne({ where: { name: 'sector' } });
+
+    if (!filter) {
+      throw new Error('Missing filter with name "sector"');
+    }
+
+    await use(filter);
+  },
+});
+
+test.describe.configure({ mode: 'serial' });
 
 test.describe('Explore E2E', () => {
-  let testManager: E2eTestManager
-  let page: Page
-  let sections: Section[]
-
-  test.beforeAll(async ({ browser }) => {
-    page = await browser.newPage()
-    testManager = await E2eTestManager.load(page)
-  })
-
-  test.beforeEach(async () => {
-    await testManager.clearDatabase()
-
-    const entityMocks = testManager.mocks()
-    sections = [
-      await entityMocks.createSection({
-        slug: 'section-1',
-        name: 'Section 1',
-        description: ':)',
-        // TODO: We should test the content of the sections including overview section and the widgets
-        // when API is returning widget data in test environment.
-        order: 2,
-        baseWidgets: []
-      }),
-      await entityMocks.createSection({
-        slug: 'section-2',
-        name: 'Section 2',
-        description: ':)',
-        order: 3,
-        baseWidgets: []
-      })
-    ]
-  })
-
-  test.afterEach(async () => {
-    await testManager.clearDatabase()
-  })
-
-  test.afterAll(async () => {
-    await testManager.close()
-  })
-
-  test('List all sections in the sidebar and main body', async () => {
-    await page.goto('/explore')
-    await expect(page.locator('#sections-container').locator('section')).toHaveCount(
-      sections.length
-    )
-    await expect(page.locator('#sidebar-sections-list').locator('a')).toHaveCount(sections.length)
-  })
-})
+  test.describe('Section Navigation', () => {
+    test('List all sections as anchor links and section elements', async ({
+      page,
+      sections,
+    }) => {
+      const count = sections.length;
+
+      await page.goto('/explore');
+      await expect(
+        page.locator('#sidebar-sections-list').locator('a'),
+      ).toHaveCount(count);
+      await expect(
+        page.locator('#in-page-sections-list').locator('a'),
+      ).toHaveCount(count);
+      await expect(
+        page.locator('#sections-container').locator('section'),
+      ).toHaveCount(count);
+    });
+
+    test('Should scroll to the correct section when clicking anchor links', async ({
+      page,
+    }) => {
+      await page.goto('/explore');
+      const sections = await page
+        .locator('#sidebar-sections-list')
+        .locator('a')
+        .all();
+
+      for (const link of sections) {
+        const href = await link.getAttribute('href');
+        const targetSectionId = href?.replace('#', '');
+
+        await link.click();
+        await expect(page).toHaveURL(new RegExp(`${href}$`));
+
+        const targetSection = page.locator(`#${targetSectionId}`);
+        await expect(targetSection).toBeInViewport();
+      }
+    });
+
+    test('Should handle direct anchor navigation and browser history', async ({
+      page,
```

Comment:
Just to make sure we are all aware of this:

Inside of a test, if you make use of the `page` object that's being passed to the test callback, you are actually using a different instance than the one that is injected in the test manager, meaning that, for example, if you use the login helper, the session exists only in that instance, and here the test driver would open a new tab/window with a brand new session.

This might not be an issue here, but just for you to know. To avoid any weird false negatives, make use of the global `page` instance, or the one that you can get from the test manager (which references the same instance) unless, of course, you actually need to work with 2 different browser for some reason